tratto/src/main/java/star/tratto/util/FileUtils.java

```diff
@@ -0,0 +1,199 @@
+import java.util.Objects;
+
+
+/**
+ * This class manages file I/O utilities.
+ */
+public class FileUtils {
+    private static final Logger logger = LoggerFactory.getLogger(FileUtils.class);
+
+    public static void appendToFile(
+            String dirPath,
+            String fileName,
+            FileFormat fileFormat,
+            String projectName,
+            Object content
+    ) {
+        try {
+            // create a new file
+            File file = createFile(Paths.get(dirPath, projectName).toString(), fileName, fileFormat);
+            // depending on the file format, save the content, accordingly
+            switch (fileFormat) {
+                case TXT, CSV -> {
+                    // Convert content to string
+                    String fileContent = (String) content;
+                    // Create a FileWriter object with append flag set to true
+                    FileWriter writer = new FileWriter(file, true);
+                    // Write content to the file
+                    writer.write(fileContent + "\n");
+                    // Close the FileWriter object
+                    writer.close();
+                }
+                case JSON -> {
+                    ObjectMapper objectMapper = new ObjectMapper();
+                    objectMapper.writerWithDefaultPrettyPrinter().writeValue(file, content);
+                }
+                default -> {
+                    String errMsg = String.format("File format %s not yet supported to save the content of a file.", fileFormat.getValue());
+                    logger.error(errMsg);
+                }
+            }
+        } catch (IOException | FolderCreationFailedException | FileNotCreatedException e) {
+            e.printStackTrace();
+        }
+    }
+
+    public static void appendToFileExclusive(
+            String dirPath,
+            String fileName,
+            FileFormat fileFormat,
+            String projectName,
+            String content
+    ) {
+        String filePath = Paths.get(dirPath, projectName, fileName + fileFormat.getValue()).toString();
+        File file = new File(filePath);
+        boolean found = false;
+        if (file.exists()) {
+            try (BufferedReader br = new BufferedReader(new FileReader(filePath))) {
+                String line;
+                while ((line = br.readLine()) != null) {
+                    if (line.contains(content)) {
+                        found = true;
+                        break;
+                    }
+                }
+            } catch (IOException e) {
+                logger.error("Error reading file: " + e.getMessage());
+            }
+        }
+        if (!found) {
+            appendToFile(
+                    dirPath,
+                    fileName,
+                    fileFormat,
+                    projectName,
+                    content
+            );
+        }
+    }
+
+    /**
+     * The method creates a file {@link File} within a given directory.
```

Comment:
A `File` is a representation of a pathname.  It can exist with or without there being a corresponding file in the file system.  So this comment conflates two independent concepts, and from the documentation I'm not sure exactly what it does.